### PR TITLE
Implement cohort extraction endpoint

### PR DESCRIPTION
**Pull request by juliangruendner**

### CODE REVIEW

**Review by alexanderkiel — CHANGES_REQUESTED** [review]

**Comment by alexanderkiel** [comment]
Are we sure that returning a JSON array with potentially 1 million ID's is feasible?

**Comment by juliangruendner** [comment]
> Are we sure that returning a JSON array with potentially 1 million ID's is feasible?

@alexanderkiel , as discussed - no -> paging will most likely be have to added in the next months:
https://github.com/medizininformatik-initiative/flare/issues/180

File: src/test/java/de/medizininformatikinitiative/flare/rest/QueryControllerTest.java [hunk 1]
```diff
@@ -42,8 +50,8 @@ void setUp() {
     }
 
     @Test
-    void execute() {
-        when(queryService.execute(STRUCTURED_QUERY)).thenReturn(Mono.just(1));
+    void executeCohort() {
```
Comment (alexanderkiel) [inline src/test/java/de/medizininformatikinitiative/flare/rest/QueryControllerTest.java:53]: this is execute

File: src/test/java/de/medizininformatikinitiative/flare/rest/QueryControllerTest.java [hunk 2]
```diff
@@ -75,6 +83,43 @@ void execute() {
                 .expectBody().json("1");
     }
 
+
+    @Test
+    void execute() throws JsonProcessingException {
```
Comment (alexanderkiel) [inline src/test/java/de/medizininformatikinitiative/flare/rest/QueryControllerTest.java:88]: this is executeCohort

File: src/test/java/de/medizininformatikinitiative/flare/service/StructuredQueryServiceIT.java [hunk 3]
```diff
@@ -131,25 +135,23 @@ void setUp() throws URISyntaxException, IOException {
     void execute_Criterion() {
         var query = StructuredQuery.of(CriterionGroup.of(CriterionGroup.of(Criterion.of(ContextualConcept.of(DIAGNOSIS, Concept.of(I08))))));
 
-        var result = service.execute(query).block();
+        var result = getExecutionResult(query);
```
Comment (alexanderkiel) [inline src/test/java/de/medizininformatikinitiative/flare/service/StructuredQueryServiceIT.java:138]: Sorry, I did see this in my last review. But we have to test the returned ID's here and not only the size.

File: src/test/java/de/medizininformatikinitiative/flare/service/StructuredQueryServiceIT.java [hunk 4]
```diff
@@ -131,25 +135,23 @@ void setUp() throws URISyntaxException, IOException {
...
 
         assertThat(result).isOne();
```
Comment (alexanderkiel) [inline src/test/java/de/medizininformatikinitiative/flare/service/StructuredQueryServiceIT.java:140]: Also please use StepVerifier here. The use of `block()` surely did cam from a time were I didn't know about StepVerifier.

File: src/test/java/de/medizininformatikinitiative/flare/service/StructuredQueryServiceTest.java [hunk 5]
```diff
@@ -85,7 +89,7 @@ class SingleExpansion {
             void execute() {
                 var query = query(incl(PATIENT));
 
-                var result = service.execute(query).block();
+                var result = getExecutionResult(query);
```
Comment (alexanderkiel) [inline src/test/java/de/medizininformatikinitiative/flare/service/StructuredQueryServiceTest.java:92]: same as for the IT

**Review by alexanderkiel — CHANGES_REQUESTED** [review]

**Pull request merged**